tutorials/dataframe/df106_HiggsToFourLeptons.C

```diff
@@ -0,0 +1,284 @@
+   auto df_other = df_mc.Filter("sample_category == \"other\"")
+                      .Histo1D<float>(ROOT::RDF::TH1DModel("other", "m4l", 24, 80, 170), "m4l", "weight");
+
+   // Book the invariant mass histogram for the data
+   auto df_h_mass_data = df_analysis.Filter("reweighting == false")
+                            .Filter("sample_category == \"data\"")
+                            .Define("weight_", []() { return 1; })
+                            .Histo1D<float>(ROOT::RDF::TH1DModel("data", "m4l", 24, 80, 170), "m4l", "weight_");
+
+   // Evaluate the systematic uncertainty
+
+   // The systematic uncertainty in this analysis is the MC scale factor uncertainty that depends on lepton
+   // kinematics such as pT or pseudorapidity.
+   // Muons uncertainties are negligible, as stated in https://atlas.web.cern.ch/Atlas/GROUPS/PHYSICS/PAPERS/MUON
+   // Electrons uncertainties are evaluated based on the plots available in https://doi.org/10.48550/arXiv.1908.0
+   // The uncertainties are linearly interpolated to cover a range of pT values covered by the analysis.
+   const std::vector<double> x{5.50e3, 5.52e3, 12.54e3, 17.43e3, 22.40e3, 27.48e3, 30e3, 10000e3};
+   const std::vector<double> y{0.06628, 0.06395, 0.06396, 0.03372, 0.02441, 0.01403, 0, 0};
+   unsigned int N = x.size();
+   ROOT::Math::Interpolator inter(N, ROOT::Math::Interpolation::kLINEAR);
+   inter.SetData(x, y);
+
+   //  Use the Vary method to add the systematic variations to the total MC scale factor ("weight") of the analysis
+   //  The input consists of the input column to be varied and the lambda function to compute the systematic variations.
+   //  The new output columns contain the varied values of the input column.
+   auto df_with_variations_mc =
+      df_mc
+         .Vary("weight",
+               [&inter](double x, const RVecF &pt, const RVec<unsigned int> &type) {
+                  const auto v = Mean(Map(pt[type == 11], [&inter](auto p) { return inter.Eval(p); }));
+                  return RVec<double>{(1 + v) * x, (1 - v) * x};
+               },
+               {"weight", "goodlep_pt", "goodlep_type"}, {"up", "down"})
+         .Histo1D<float>(ROOT::RDF::TH1DModel("Invariant Mass", "m4l", 24, 80, 170), "m4l", "weight");
+
+   // Create the total MC scale factor histograms: "nominal", "weight:up" and "weight:down".
+   auto histos_mc = VariationsFor(df_with_variations_mc);
+
+   // Evaluate the total MC uncertainty based on the variations. Note, in this case the uncertainties are symmetric.
+   for (unsigned int i = 0; i < histos_mc["nominal"].GetXaxis()->GetNbins(); i++) {
+      histos_mc["nominal"].SetBinError(
+         i, (histos_mc["weight:up"].GetBinContent(i) - histos_mc["nominal"].GetBinContent(i)));
+   }
+
+   // Make the plot of the data, individual MC contributions and the total MC scale factor systematic variations.
+   gROOT->SetStyle("ATLAS");
+
+   // Create canvas with pad
+   auto c = new TCanvas("c", " ", 600, 600);
```

Comment:
the saved figure looks ok, but there is a problem with the interactive mode - it immediately shuts down with the new web canvas version (while you can keep it open with the "new" version)

Reply:
> auto c = std::make_unique<TCanvas>()

at that point it can be a `TCanvas c` :grimacing: 

> it immediately shuts down with the new web canvas version

if the old graphics work fine but the new graphics do not, sounds like a problem with the new graphics :/

Reply:
the interactive session also wouldn't appear when I do it with "--web-off" option and the "TCanvas c("c", " ", 600, 600);" version

Reply:
ah yeah this is C++, I thought it was under `python -i` sorry :man_facepalming: 

yes, if the canvas is destroyed at the end of the macro it cannot be displayed after the macro has finished running. I don't know of a way around that other than using the default canvas or, if you have to define your own, "leak" it (that is, let ROOT manage its lifetime)

Reply:
and by default canvas do you mean the way I originally had it: `auto c = new TCanvas("c", " ", 600, 600);` ? or something else? or is it better to change it to the nicer `TCanvas c("c", " ", 600, 600);` but then not be able to run in interactive mode? (pure `TCanvas c;` will also not work in the interactive mode)

Reply:
I'll leave the `auto c = new TCanvas("c", " ", 600, 600);` here as I think people may want to work with it interactively too, but we can raise a general point it's not the nicest way to do it